### PR TITLE
Id system

### PR DESCRIPTION
**Pull request by Krishnan-Priyanshu**
##### SUMMARY

Relates OR Closes #0000

##### ISSUE TYPE

- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

#### Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note

```

#### Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

#### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

### CODE REVIEW

**Comment by terraform-providers-svc** [comment]
Can one of the admins verify this patch?

File: redfish/provider/common.go [hunk 1]
```diff
@@ -290,7 +315,7 @@ func (p powerOperator) PowerOperation(resetType string, maximumWaitTime int64, c
 }
 
 // Check checks iDRAC server status after provided interval until the provided timeout time
-func (s *ServerStatusChecker) Check(ctx context.Context) error {
+func (s *ServerStatusChecker) Check(ctx context.Context, sysid string) error {
```
Comment (rounak-adhikary) [inline redfish/provider/common.go:318]: This particular function does not care about `sysid` at all... it only checks that the API is working...
So we shd simply use `getSystemResource(s.Service, "")` inside this function.

File: redfish/provider/common.go [hunk 2]
```diff
@@ -212,10 +237,10 @@ type powerOperator struct {
 // server's power state for updates. The last is a pointer to a gofish.Service object with which the function can
 // interact with the server. It will return a tuple consisting of the server's power state at time of return and
 // diagnostics
-func (p powerOperator) PowerOperation(resetType string, maximumWaitTime int64, checkInterval int64) (redfish.PowerState, error) {
+func (p powerOperator) PowerOperation(sysid string, resetType string, maximumWaitTime int64, checkInterval int64) (redfish.PowerState, error) {
```
Comment (rounak-adhikary) [inline redfish/provider/common.go:240]: Probably better to add `sysid` as a field in the powerOperator struct itself.

File: redfish/provider/resource_redfish_idrac_firmware_update.go [hunk 3]
```diff
@@ -209,6 +209,15 @@ func idracFirmwareUpdateSchema() map[string]schema.Attribute {
 			Optional: true,
 			Default:  booldefault.StaticBool(true),
 		},
+		"system_id": schema.StringAttribute{
```
Comment (Sakshi-dell) [inline redfish/provider/resource_redfish_idrac_firmware_update.go:212]: check formatting

File: redfish/provider/resource_redfish_boot_sources_override_test.go [hunk 4]
```diff
@@ -71,7 +71,7 @@ func testAccRedfishResourceBootSourceLegacyconfig(testingInfo TestingServerCrede
 			endpoint = "https://%s"
 			ssl_insecure = true
 		}
-	   
+	    system_id = "System.Embedded.1"
```
Comment (Sakshi-dell) [inline redfish/provider/resource_redfish_boot_sources_override_test.go:74]: check formatting

Comment (Krishnan-Priyanshu) [reply redfish/provider/resource_redfish_boot_sources_override_test.go:74]: go formatting is already done on the file, let me know if you expect some other formatting.

**Review by rounak-adhikary — APPROVED** [review]

**Review by Sakshi-dell — APPROVED** [review]

**Pull request merged**